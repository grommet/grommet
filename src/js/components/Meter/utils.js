import { normalizeColor } from '../../utils';

export const strokeProps = (color, theme) => {
  const result = {};
  if (color) {
    if (typeof color === 'object') {
      result.stroke = normalizeColor(color.color, theme);
      if (color.opacity) {
        result.strokeOpacity = `${
          color.opacity === true
            ? theme.global.opacity.medium
            : theme.global.opacity[color.opacity] || color.opacity
        }`;
      }
    } else {
      result.stroke = normalizeColor(color, theme);
    }
  }
  return result;
};

export const fillProps = (color, theme) => {
  const result = {};
  if (color) {
    if (typeof color === 'object') {
      result.fill = normalizeColor(color.color, theme);
      // if (color.opacity) {
      //   result.strokeOpacity = `${
      //     color.opacity === true
      //       ? theme.global.opacity.medium
      //       : theme.global.opacity[color.opacity] || color.opacity
      //   }`;
      // }
    } else {
      result.fill = normalizeColor(color, theme);
    }
  }
  return result;
};

const neutralExp = /^neutral-\d+/;

export const defaultColor = (index, theme, valuesLength) => {
  if (index === valuesLength - 1 && theme.meter.color) {
    return theme.meter.color;
  }
  // We want the last value to have the first color
  const colorIndex = valuesLength - index - 1;
  if (theme.meter && theme.meter.colors) {
    const colors =
      theme.meter.colors[theme.dark ? 'dark' : 'light'] || theme.meter.colors;
    return colors[colorIndex % colors.length];
  }
  const colors = Object.keys(theme.global.colors).filter(n =>
    n.match(/^graph-[0-9]$/),
  );
  if (colors.length > 0) {
    return colors[colorIndex % colors.length];
  }
  // Deprecate using "neutral-*" color names. Remove eventually.
  const neutralColors = Object.keys(theme.global.colors).filter(k =>
    neutralExp.test(k),
  );
  return neutralColors[colorIndex % neutralColors.length];
};
