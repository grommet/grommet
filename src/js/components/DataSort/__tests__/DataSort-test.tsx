import React from 'react';
import {
  act,
  fireEvent,
  render,
  screen,
  waitFor,
} from '@testing-library/react';
import 'jest-styled-components';

import { Data } from '../../Data';
import { DataFilters } from '../../DataFilters';
import { Grommet } from '../../Grommet';
import { DataSort } from '..';
import { createPortal, expectPortal } from '../../../utils/portal';

const data = [{ name: 'a' }, { name: 'b' }];

describe('DataSort', () => {
  window.scrollTo = jest.fn();
  beforeEach(createPortal);

  test('renders', () => {
    const { container } = render(
      <Grommet>
        <Data data={data}>
          <DataFilters>
            <DataSort />
          </DataFilters>
        </Data>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('drop', () => {
    jest.useFakeTimers();

    const { container, getByRole } = render(
      <Grommet>
        <Data id="test-data" data={data}>
          <DataSort drop />
        </Data>
      </Grommet>,
    );

    expect(getByRole('button', { name: 'Open sort' })).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();

    fireEvent.click(getByRole('button', { name: 'Open sort' }));
    // advance timers so drop can open
    act(() => jest.advanceTimersByTime(200));

    // snapshot on drop
    expectPortal('test-data--sort-control').toMatchSnapshot();
  });

  test('sort when data array is empty', () => {
    const { container } = render(
      <Grommet>
        <Data id="test-data" data={[]}>
          <DataSort />
        </Data>
      </Grommet>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render tooltip', async () => {
    render(
      <Grommet>
        <Data data={data}>
          <DataSort drop />
        </Data>
      </Grommet>,
    );

    const sortButton = screen.getByRole('button', { name: 'Open sort' });
    fireEvent.mouseOver(sortButton);

    const tooltip = await waitFor(() => screen.getByText('Open sort'));

    expect(tooltip).toBeTruthy();
  });

  test('properties', async () => {
    const items = [
      { name: 'a', loc: 'Home' },
      { name: 'b', loc: 'School' },
    ];
    const properties = {
      name: { label: 'My Name' },
      loc: { label: 'Location' },
    };

    render(
      <Grommet>
        <Data data={items} properties={properties}>
          <DataFilters>
            <DataSort />
          </DataFilters>
        </Data>
      </Grommet>,
    );

    const propButton = screen.getByRole('button', { name: 'Open Drop' });
    fireEvent.click(propButton);
    const drop = await waitFor(() => screen.getByRole('listbox'));

    expect(drop).toBeTruthy();
    expect(drop).toMatchSnapshot();
  });
});
