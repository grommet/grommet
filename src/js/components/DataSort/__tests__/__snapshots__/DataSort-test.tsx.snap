// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSort drop 1`] = `
.c1 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c3 {
  display: inline-block;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c3 g {
  fill: inherit;
  stroke: inherit;
}

.c3 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c3 *[stroke*='#'],
.c3 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c3 *[fill-rule],
.c3 *[FILL-RULE],
.c3 *[fill*='#'],
.c3 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c2 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
  line-height: 0;
  padding: 12px;
}

.c2:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c2:focus >circle,
.c2:focus >ellipse,
.c2:focus >line,
.c2:focus >path,
.c2:focus >polygon,
.c2:focus >polyline,
.c2:focus >rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c2:focus ::-moz-focus-inner {
  border: 0;
}

.c2:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c2:focus:not(:focus-visible) >circle,
.c2:focus:not(:focus-visible) >ellipse,
.c2:focus:not(:focus-visible) >line,
.c2:focus:not(:focus-visible) >path,
.c2:focus:not(:focus-visible) >polygon,
.c2:focus:not(:focus-visible) >polyline,
.c2:focus:not(:focus-visible) >rect {
  outline: none;
  box-shadow: none;
}

.c2:focus:not(:focus-visible) ::-moz-focus-inner {
  border: 0;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

<div
  class="c0"
>
  <div
    class="c1"
    id="test-data"
  >
    <button
      aria-label="Open sort"
      class="c2"
      id="test-data--sort-control"
      type="button"
    >
      <svg
        aria-label="Descend"
        class="c3"
        viewBox="0 0 24 24"
      >
        <path
          d="m2 16 6 6 6-6M11 3h11M11 7h8m-8 4h5M8 22V2"
          fill="none"
          stroke="#000"
          stroke-width="2"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`DataSort drop 2`] = `
<button
  aria-label="Open sort"
  class="StyledButton-sc-323bzc-0 geHpTD"
  data-g-tabindex="none"
  id="test-data--sort-control"
  tabindex="-1"
  type="button"
>
  <svg
    aria-label="Descend"
    class="StyledIcon-sc-ofa7kd-0 bqRbRY"
    viewBox="0 0 24 24"
  >
    <path
      d="m2 16 6 6 6-6M11 3h11M11 7h8m-8 4h5M8 22V2"
      fill="none"
      stroke="#000"
      stroke-width="2"
    />
  </svg>
</button>
`;

exports[`DataSort drop 3`] = `
"@media only screen and (max-width: 768px) {
  .lAiqC {
    margin-bottom: 6px;
  }
}
@media only screen and (max-width: 768px) {
  .eYrCJw {
    border-bottom: solid 1px rgba(0, 0, 0, 0.33);
  }
}
@media only screen and (max-width: 768px) {
  .iBKXSA {
    margin-left: 6px;
    margin-right: 6px;
  }
}
@media only screen and (max-width: 768px) {
  .efNsjV {
    border-bottom: solid 1px rgba(0, 0, 0, 0.33);
  }
}
@media only screen and (max-width: 768px) {
  .efNsjV {
    padding: 6px;
  }
}
@media only screen and (max-width: 768px) {
  .irUZj {
    margin-right: 6px;
  }
}
@media only screen and (max-width: 768px) {
  .injqDh {
    border: solid 2px rgba(0, 0, 0, 0.15);
  }
}
@media only screen and (max-width: 768px) {
  .eYqvdH {
    margin-top: 12px;
  }
}
@media only screen and (max-width: 768px) {
  .cvOsLl {
    height: 6px;
  }
}"
`;

exports[`DataSort properties 1`] = `
.c3 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: flex-start;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  padding: 12px;
}

.c5 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  height: 0px;
  flex: 0 0 auto;
}

.c4 {
  margin: 0px;
  font-size: 18px;
  line-height: 24px;
}

.c1 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c1:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c1:focus:not(:focus-visible) >circle,
.c1:focus:not(:focus-visible) >ellipse,
.c1:focus:not(:focus-visible) >line,
.c1:focus:not(:focus-visible) >path,
.c1:focus:not(:focus-visible) >polygon,
.c1:focus:not(:focus-visible) >polyline,
.c1:focus:not(:focus-visible) >rect {
  outline: none;
  box-shadow: none;
}

.c1:focus:not(:focus-visible) ::-moz-focus-inner {
  border: 0;
}

.c0 {
  position: relative;
  scroll-behavior: smooth;
  overflow: auto;
  outline: none;
}

.c2 {
  display: block;
  width: 100%;
}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 768px) {
  .c3 {
    padding: 6px;
  }
}

@media only screen and (max-width: 768px) {

}

@media screen and (-ms-high-contrast:active),(-ms-high-contrast:none) {

}

@media screen and (-ms-high-contrast:active),(-ms-high-contrast:none) {

}

<div
  class="c0"
  role="listbox"
  tabindex="-1"
>
  <button
    aria-posinset="1"
    aria-selected="false"
    aria-setsize="2"
    class="c1 c2"
    role="option"
    tabindex="-1"
    type="button"
  >
    <div
      class="c3"
    >
      <span
        class="c4"
      >
        Location
      </span>
    </div>
  </button>
  <button
    aria-posinset="2"
    aria-selected="false"
    aria-setsize="2"
    class="c1 c2"
    role="option"
    tabindex="-1"
    type="button"
  >
    <div
      class="c3"
    >
      <span
        class="c4"
      >
        My Name
      </span>
    </div>
  </button>
  <div
    class="c5"
  />
</div>
`;

exports[`DataSort renders 1`] = `
.c1 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c3 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  height: 100%;
}

.c4 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 1 1;
  overflow: auto;
}

.c5 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c7 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border-bottom: solid 1px rgba(0, 0, 0, 0.33);
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c9 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: center;
  min-width: 0;
  min-height: 0;
  flex-direction: row;
  justify-content: space-between;
}

.c10 {
  display: flex;
  box-sizing: border-box;
  min-width: 0;
  min-height: 0;
  flex-direction: row;
  flex: 1 1;
  flex-basis: auto;
}

.c14 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-left: 12px;
  margin-right: 12px;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c16 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border-bottom: solid 1px rgba(0, 0, 0, 0.33);
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  padding: 12px;
}

.c17 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c19 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-right: 12px;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c21 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: center;
  border: solid 2px rgba(0, 0, 0, 0.15);
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  height: 24px;
  width: 24px;
  justify-content: center;
  border-radius: 100%;
}

.c23 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-top: 24px;
  align-items: center;
  min-width: 0;
  min-height: 0;
  flex-direction: row;
  flex: 0 0 auto;
  justify-content: space-between;
}

.c22 {
  flex: 0 0 auto;
  align-self: stretch;
  height: 12px;
}

.c15 {
  display: inline-block;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #7D4CDB;
  stroke: #7D4CDB;
}

.c15 g {
  fill: inherit;
  stroke: inherit;
}

.c15 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c15 *[stroke*='#'],
.c15 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c15 *[fill-rule],
.c15 *[FILL-RULE],
.c15 *[fill*='#'],
.c15 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c6 {
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 6px;
  margin-bottom: 6px;
  font-size: 18px;
  line-height: 24px;
}

.c8 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c8:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c8:focus:not(:focus-visible) >circle,
.c8:focus:not(:focus-visible) >ellipse,
.c8:focus:not(:focus-visible) >line,
.c8:focus:not(:focus-visible) >path,
.c8:focus:not(:focus-visible) >polygon,
.c8:focus:not(:focus-visible) >polyline,
.c8:focus:not(:focus-visible) >rect {
  outline: none;
  box-shadow: none;
}

.c8:focus:not(:focus-visible) ::-moz-focus-inner {
  border: 0;
}

.c24 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #7D4CDB;
  color: #f8f8f8;
  border-radius: 18px;
  transition-property: color,background-color,border-color,box-shadow;
  transition-duration: 0.1s;
  transition-timing-function: ease-in-out;
}

.c24:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c24:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c24:focus >circle,
.c24:focus >ellipse,
.c24:focus >line,
.c24:focus >path,
.c24:focus >polygon,
.c24:focus >polyline,
.c24:focus >rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c24:focus ::-moz-focus-inner {
  border: 0;
}

.c24:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c24:focus:not(:focus-visible) >circle,
.c24:focus:not(:focus-visible) >ellipse,
.c24:focus:not(:focus-visible) >line,
.c24:focus:not(:focus-visible) >path,
.c24:focus:not(:focus-visible) >polygon,
.c24:focus:not(:focus-visible) >polyline,
.c24:focus:not(:focus-visible) >rect {
  outline: none;
  box-shadow: none;
}

.c24:focus:not(:focus-visible) ::-moz-focus-inner {
  border: 0;
}

.c2 {
  max-width: 100%;
}

.c18 {
  display: flex;
  flex-direction: row;
  align-items: center;
  user-select: none;
  width: fit-content;
  cursor: pointer;
}

.c18:hover input:not([disabled])+div,
.c18:hover input:not([disabled])+span {
  border-color: #000000;
}

.c20 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
  cursor: pointer;
}

.c12 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0, 0, 0, 0.33);
  border-radius: 4px;
  border: none;
  outline: none;
  border: none;
}

.c12::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c12::-moz-placeholder {
  color: #AAAAAA;
}

.c12:-ms-input-placeholder {
  color: #AAAAAA;
}

.c12 ::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c12::-moz-focus-inner {
  border: none;
  outline: none;
}

.c12:-moz-placeholder,
.c12::-moz-placeholder {
  opacity: 1;
}

.c11 {
  position: relative;
  width: 100%;
}

.c13 {
  cursor: pointer;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

@media only screen and (max-width: 768px) {
  .c5 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width: 768px) {
  .c7 {
    border-bottom: solid 1px rgba(0, 0, 0, 0.33);
  }
}

@media only screen and (max-width: 768px) {
  .c14 {
    margin-left: 6px;
    margin-right: 6px;
  }
}

@media only screen and (max-width: 768px) {
  .c16 {
    border-bottom: solid 1px rgba(0, 0, 0, 0.33);
  }
}

@media only screen and (max-width: 768px) {
  .c16 {
    padding: 6px;
  }
}

@media only screen and (max-width: 768px) {
  .c19 {
    margin-right: 6px;
  }
}

@media only screen and (max-width: 768px) {
  .c21 {
    border: solid 2px rgba(0, 0, 0, 0.15);
  }
}

@media only screen and (max-width: 768px) {
  .c23 {
    margin-top: 12px;
  }
}

@media only screen and (max-width: 768px) {
  .c22 {
    height: 6px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    id="data"
  >
    <form
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <div
            class="c1"
          >
            <div
              class="c5 "
            >
              <label
                class="c6"
                for="data--sort-property"
              >
                Sort by
              </label>
              <div
                class="c7 "
              >
                <button
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-label="Open Drop"
                  class="c8 "
                  id="data--sort-property"
                  type="button"
                >
                  <div
                    class="c9"
                  >
                    <div
                      class="c10"
                    >
                      <div
                        class="c11"
                      >
                        <input
                          autocomplete="off"
                          class="c12 c13"
                          id="data--sort-property__input"
                          inert=""
                          name="_sort.property"
                          readonly=""
                          tabindex="-1"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="c14"
                      style="min-width: auto;"
                    >
                      <svg
                        aria-label="FormDown"
                        class="c15"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="m18 9-6 6-6-6"
                          fill="none"
                          stroke="#000"
                          stroke-width="2"
                        />
                      </svg>
                    </div>
                  </div>
                </button>
              </div>
            </div>
            <div
              class="c5 "
            >
              <label
                class="c6"
                for="data--sort-direction"
              >
                Sort direction
              </label>
              <div
                class="c16 "
              >
                <div
                  class="c17"
                  id="data--sort-direction"
                  role="radiogroup"
                >
                  <label
                    class="c18"
                  >
                    <div
                      class="c19 "
                    >
                      <input
                        class="c20"
                        name="_sort.direction"
                        tabindex="0"
                        type="radio"
                        value="asc"
                      />
                      <div
                        class="c21 "
                      />
                    </div>
                    <span
                      class="StyledRadioButton__StyledRadioButtonLabel-sc-g1f6ld-2"
                    >
                      Ascending
                    </span>
                  </label>
                  <div
                    class="c22"
                  />
                  <label
                    class="c18"
                  >
                    <div
                      class="c19 "
                    >
                      <input
                        class="c20"
                        name="_sort.direction"
                        tabindex="-1"
                        type="radio"
                        value="desc"
                      />
                      <div
                        class="c21 "
                      />
                    </div>
                    <span
                      class="StyledRadioButton__StyledRadioButtonLabel-sc-g1f6ld-2"
                    >
                      Descending
                    </span>
                  </label>
                </div>
              </div>
            </div>
          </div>
        </div>
        <footer
          class="c23"
        >
          <button
            class="c24"
            type="submit"
          >
            Apply filters
          </button>
        </footer>
      </div>
    </form>
  </div>
</div>
`;

exports[`DataSort sort when data array is empty 1`] = `
.c1 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c3 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c5 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border-bottom: solid 1px rgba(0, 0, 0, 0.33);
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c7 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: center;
  min-width: 0;
  min-height: 0;
  flex-direction: row;
  justify-content: space-between;
}

.c8 {
  display: flex;
  box-sizing: border-box;
  min-width: 0;
  min-height: 0;
  flex-direction: row;
  flex: 1 1;
  flex-basis: auto;
}

.c12 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-left: 12px;
  margin-right: 12px;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c14 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border-bottom: solid 1px rgba(0, 0, 0, 0.33);
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  padding: 12px;
}

.c15 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c17 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-right: 12px;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  flex: 0 0 auto;
}

.c19 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  align-items: center;
  border: solid 2px rgba(0, 0, 0, 0.15);
  min-width: 0;
  min-height: 0;
  flex-direction: column;
  height: 24px;
  width: 24px;
  justify-content: center;
  border-radius: 100%;
}

.c20 {
  flex: 0 0 auto;
  align-self: stretch;
  height: 12px;
}

.c13 {
  display: inline-block;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #7D4CDB;
  stroke: #7D4CDB;
}

.c13 g {
  fill: inherit;
  stroke: inherit;
}

.c13 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c13 *[stroke*='#'],
.c13 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c13 *[fill-rule],
.c13 *[FILL-RULE],
.c13 *[fill*='#'],
.c13 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c4 {
  margin-left: 12px;
  margin-right: 12px;
  margin-top: 6px;
  margin-bottom: 6px;
  font-size: 18px;
  line-height: 24px;
}

.c6 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  color: inherit;
  outline: none;
  border: none;
  padding: 0;
  text-align: inherit;
}

.c6:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c6:focus:not(:focus-visible) >circle,
.c6:focus:not(:focus-visible) >ellipse,
.c6:focus:not(:focus-visible) >line,
.c6:focus:not(:focus-visible) >path,
.c6:focus:not(:focus-visible) >polygon,
.c6:focus:not(:focus-visible) >polyline,
.c6:focus:not(:focus-visible) >rect {
  outline: none;
  box-shadow: none;
}

.c6:focus:not(:focus-visible) ::-moz-focus-inner {
  border: 0;
}

.c2 {
  max-width: 100%;
}

.c16 {
  display: flex;
  flex-direction: row;
  align-items: center;
  user-select: none;
  width: fit-content;
  cursor: pointer;
}

.c16:hover input:not([disabled])+div,
.c16:hover input:not([disabled])+span {
  border-color: #000000;
}

.c18 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
  cursor: pointer;
}

.c10 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  width: 100%;
  padding: 11px;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0, 0, 0, 0.33);
  border-radius: 4px;
  border: none;
  outline: none;
  border: none;
}

.c10::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c10::-moz-placeholder {
  color: #AAAAAA;
}

.c10:-ms-input-placeholder {
  color: #AAAAAA;
}

.c10 ::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c10::-moz-focus-inner {
  border: none;
  outline: none;
}

.c10:-moz-placeholder,
.c10::-moz-placeholder {
  opacity: 1;
}

.c9 {
  position: relative;
  width: 100%;
}

.c11 {
  cursor: pointer;
}

.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

@media only screen and (max-width: 768px) {
  .c3 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width: 768px) {
  .c5 {
    border-bottom: solid 1px rgba(0, 0, 0, 0.33);
  }
}

@media only screen and (max-width: 768px) {
  .c12 {
    margin-left: 6px;
    margin-right: 6px;
  }
}

@media only screen and (max-width: 768px) {
  .c14 {
    border-bottom: solid 1px rgba(0, 0, 0, 0.33);
  }
}

@media only screen and (max-width: 768px) {
  .c14 {
    padding: 6px;
  }
}

@media only screen and (max-width: 768px) {
  .c17 {
    margin-right: 6px;
  }
}

@media only screen and (max-width: 768px) {
  .c19 {
    border: solid 2px rgba(0, 0, 0, 0.15);
  }
}

@media only screen and (max-width: 768px) {
  .c20 {
    height: 6px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    id="test-data"
  >
    <form
      class="c2"
    >
      <div
        class="c3 "
      >
        <label
          class="c4"
          for="test-data--sort-property"
        >
          Sort by
        </label>
        <div
          class="c5 "
        >
          <button
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Open Drop"
            class="c6 "
            id="test-data--sort-property"
            type="button"
          >
            <div
              class="c7"
            >
              <div
                class="c8"
              >
                <div
                  class="c9"
                >
                  <input
                    autocomplete="off"
                    class="c10 c11"
                    id="test-data--sort-property__input"
                    inert=""
                    name="_sort.property"
                    readonly=""
                    tabindex="-1"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="c12"
                style="min-width: auto;"
              >
                <svg
                  aria-label="FormDown"
                  class="c13"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m18 9-6 6-6-6"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </div>
            </div>
          </button>
        </div>
      </div>
      <div
        class="c3 "
      >
        <label
          class="c4"
          for="test-data--sort-direction"
        >
          Sort direction
        </label>
        <div
          class="c14 "
        >
          <div
            class="c15"
            id="test-data--sort-direction"
            role="radiogroup"
          >
            <label
              class="c16"
            >
              <div
                class="c17 "
              >
                <input
                  class="c18"
                  name="_sort.direction"
                  tabindex="0"
                  type="radio"
                  value="asc"
                />
                <div
                  class="c19 "
                />
              </div>
              <span
                class="StyledRadioButton__StyledRadioButtonLabel-sc-g1f6ld-2"
              >
                Ascending
              </span>
            </label>
            <div
              class="c20"
            />
            <label
              class="c16"
            >
              <div
                class="c17 "
              >
                <input
                  class="c18"
                  name="_sort.direction"
                  tabindex="-1"
                  type="radio"
                  value="desc"
                />
                <div
                  class="c19 "
                />
              </div>
              <span
                class="StyledRadioButton__StyledRadioButtonLabel-sc-g1f6ld-2"
              >
                Descending
              </span>
            </label>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
